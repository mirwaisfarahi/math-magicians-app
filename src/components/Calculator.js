import React, { Component } from 'react';
import { Container, Row, Col } from 'reactstrap';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Container className="calculator">
        <Row className="display-box">
          <Col className="bg-secondary border result">0</Col>
        </Row>
        <Row>
          <Col className="light column border">AC</Col>
          <Col className="light column border">+/-</Col>
          <Col className="light column border">%</Col>
          <Col className="orange column border">/</Col>
        </Row>
        <Row>
          <Col className="light column border">7</Col>
          <Col className="light column border">8</Col>
          <Col className="light column border">9</Col>
          <Col className="orange column border">*</Col>
        </Row>
        <Row>
          <Col className="light column border">4</Col>
          <Col className="light column border">5</Col>
          <Col className="light column border">6</Col>
          <Col className="orange column border">-</Col>
        </Row>
        <Row>
          <Col className="light column border">1</Col>
          <Col className="light column border">2</Col>
          <Col className="light column border">3</Col>
          <Col className="orange column border">+</Col>
        </Row>
        <Row>
          <Col className="light column border">0</Col>
          <Col className="light">
            <Row>
              <Col className="light column border">.</Col>
              <Col className="orange column border">=</Col>
            </Row>
          </Col>
        </Row>
      </Container>
    );
  }
}

export default Calculator;
