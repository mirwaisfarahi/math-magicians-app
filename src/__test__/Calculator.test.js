import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../components/Calculator';

describe('test calculator function', () => {
  it('Match Calculator DOM', () => {
    const tree = render(<Calculator />);
    expect(tree).toMatchSnapshot();
  });
});
