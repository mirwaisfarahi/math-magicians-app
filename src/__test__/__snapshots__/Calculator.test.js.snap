// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test calculator function Match Calculator DOM 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculatorHeader"
      >
        <h4>
          Let's do some math!
        </h4>
        <div
          class="calculator container"
        >
          <div
            class="display-box row"
          >
            <div
              class="bg-secondary border result col"
            >
              <span>
                 
                 
              </span>
              <span>
                 
                 
              </span>
              <span>
                 
                 
              </span>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="light column border col"
            >
              AC
            </div>
            <div
              class="light column border col"
            >
              +/-
            </div>
            <div
              class="light column border col"
            >
              %
            </div>
            <div
              class="orange column border col"
            >
              รท
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="light column border col"
            >
              7
            </div>
            <div
              class="light column border col"
            >
              8
            </div>
            <div
              class="light column border col"
            >
              9
            </div>
            <div
              class="orange column border col"
            >
              x
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="light column border col"
            >
              4
            </div>
            <div
              class="light column border col"
            >
              5
            </div>
            <div
              class="light column border col"
            >
              6
            </div>
            <div
              class="orange column border col"
            >
              -
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="light column border col"
            >
              1
            </div>
            <div
              class="light column border col"
            >
              2
            </div>
            <div
              class="light column border col"
            >
              3
            </div>
            <div
              class="orange column border col"
            >
              +
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="light column border col"
            >
              0
            </div>
            <div
              class="light col"
            >
              <div
                class="row"
              >
                <div
                  class="light column border col"
                >
                  .
                </div>
                <div
                  class="orange column border col"
                >
                  =
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculatorHeader"
    >
      <h4>
        Let's do some math!
      </h4>
      <div
        class="calculator container"
      >
        <div
          class="display-box row"
        >
          <div
            class="bg-secondary border result col"
          >
            <span>
               
               
            </span>
            <span>
               
               
            </span>
            <span>
               
               
            </span>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="light column border col"
          >
            AC
          </div>
          <div
            class="light column border col"
          >
            +/-
          </div>
          <div
            class="light column border col"
          >
            %
          </div>
          <div
            class="orange column border col"
          >
            รท
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="light column border col"
          >
            7
          </div>
          <div
            class="light column border col"
          >
            8
          </div>
          <div
            class="light column border col"
          >
            9
          </div>
          <div
            class="orange column border col"
          >
            x
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="light column border col"
          >
            4
          </div>
          <div
            class="light column border col"
          >
            5
          </div>
          <div
            class="light column border col"
          >
            6
          </div>
          <div
            class="orange column border col"
          >
            -
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="light column border col"
          >
            1
          </div>
          <div
            class="light column border col"
          >
            2
          </div>
          <div
            class="light column border col"
          >
            3
          </div>
          <div
            class="orange column border col"
          >
            +
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="light column border col"
          >
            0
          </div>
          <div
            class="light col"
          >
            <div
              class="row"
            >
              <div
                class="light column border col"
              >
                .
              </div>
              <div
                class="orange column border col"
              >
                =
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
